src/Container/Container.php

```diff
@@ -818,7 +823,10 @@ protected function resolvePrimitive( ReflectionParameter $parameter ) {
 	 */
 	protected function resolveClass( ReflectionParameter $parameter ) {
 		try {
-			return $this->make( $parameter->getClass()->name );
+			// Note: ReflectionParameter::getClass() is deprecated since PHP 8
```

Comment:
Can we also add a `@note` that `getType()` isn't supported until php7? I'd like to make it clear why we have this if/then/else logic. At first glance it seems like a quick cleanup task to just use `getType()` instead of `getClass()`, but it isn't that simple.

Reply:
Resolved e8dc6e60